haystack/utils/deepsetcloud.py

```diff
@@ -325,6 +378,24 @@ def list_pipeline_configs(self, workspace: Optional[str] = None, headers: dict =
         generator = self.client.get_with_auto_paging(url=pipelines_url, headers=headers)
         return generator
 
+    def save_pipeline_config(
+        self, config: dict, pipeline_config_name: str, workspace: Optional[str] = None, headers: dict = None
+    ):
+        config["name"] = pipeline_config_name
+        workspace_url = self._build_workspace_url(workspace=workspace)
+        pipelines_url = f"{workspace_url}/pipelines"
+        response = self.client.post(url=pipelines_url, data=yaml.dump(config), headers=headers)
+        assert response.json()["name"] == pipeline_config_name
```

Comment:
Just curious, is that to validate that the server's reply is correct? Maybe it's worth adding a small comment, it's not super-obvious what this assert is for

(same for `update_pipeline_config`)

Reply:
I changed it to a proper warning as a HTTP 200 might signal success anyway.